src/node_dotenv.cc

```diff
@@ -14,21 +14,23 @@ using v8::String;
 std::vector<std::string> Dotenv::GetPathFromArgs(
     const std::vector<std::string>& args) {
   const auto find_match = [](const std::string& arg) {
-    const std::string_view flag = "--env-file";
-    return strncmp(arg.c_str(), flag.data(), flag.size()) == 0;
+    return arg == "--" || arg.starts_with("--env-file");
   };
   std::vector<std::string> paths;
   auto path = std::find_if(args.begin(), args.end(), find_match);
 
   while (path != args.end()) {
+    if (*path == "--") {
```

Comment:
Please move this before the while loop

Reply:
I don't think this is correct, it needs to be checked inside the loop.

Checking before the loop only works if no `--env-file` node option is used.

This command should be fixed by the change

```sh
node -- --env-file nonexistent-file
```

but in this call, node will most likely still try to read the file `nonexistent-file2`:

```sh
node --env-file nonexistent-file1 -- --env-file nonexistent-file2
```

The command should only report an error for the first file, but not for the second.

```
node: nonexistent-file1: not found
```

@RedYetiDev, could you please verify my assumption?

Reply:
Yes! You are correct. Thanks for being so diligent!

By the way, I've added you as a Co-Author, as you have been a huge help getting this PR ready!

Reply:
Thanks, I'm glad I could help. Hopefully, #53060 does not reintroduce the bugs. :slightly_smiling_face: